scripts/generate_manifest.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+from argparse import ArgumentParser
+import json
+import os
+import sys
+
+
+def _parse_sample_filename(file: str) -> tuple[str, str]:  
+    # `generate_manifest()` ensures `file` ends in `.zip`
+    file, _ = os.path.splitext(file)
+    components = file.split(':')
+    if len(components) != 3:
+        raise Exception(f"Malformed sample file name: {file + '.zip'}")
+    _, package, version = components
+
+    return package.replace(',', '/'), version
+
+
+def generate_manifest(sample_dir: str) -> dict[str, list[str]]:
+    manifest = {}
+
+    for _, _, files in os.walk(sample_dir):
+        for file in files:
+            if not file.endswith(".zip"):
+                continue
+            package, version = _parse_sample_filename(file)
+            if package not in manifest:
+                manifest[package] = [version] if version else []
+            elif version and version not in manifest[package]:
+                manifest[package].append(version)
+                manifest[package].sort()
```

Comment:
nit (feel free to ignore): this is probably lexicographical sorting, do we need version-specific one? e.g.

```
from packaging.version import parse as parse_version

manifest[package].sort(key=parse_version)
```

Reply:
Yes, it's the lexicographical sorting currently.  I like this idea.